src/Tools/ExternalAccess/Razor/RazorDocumentServiceProviderWrapper.cs

```diff
@@ -1,27 +1,25 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
-#nullable enable annotations
 
 using System;
+using System.Runtime.CompilerServices;
 using Microsoft.CodeAnalysis.Host;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.ExternalAccess.Razor
 {
     internal sealed class RazorDocumentServiceProviderWrapper : IDocumentServiceProvider, IDocumentOperationService
     {
         private readonly IRazorDocumentServiceProvider _innerDocumentServiceProvider;
-        private readonly object _lock;
 
-        private RazorSpanMappingServiceWrapper? _spanMappingService;
-        private RazorDocumentExcerptServiceWrapper? _excerptService;
-        private RazorDocumentPropertiesServiceWrapper? _documentPropertiesService;
+        private StrongBox<ISpanMappingService?>? _spanMappingService;
```

Comment:
📝 `StrongBox<T>` is used because `LazyInitialization.EnsureInitialized` doesn't allow the callback to return `null`. This pattern allows us to cache both the non-null and null results.

Reply:
Ah, ok.  Null is considered an initialized result.

Reply:
Putting this comment in code would be helpful for future readers :)

Reply:
➡️ Now in a comment